Client/src/Common/Submitter/ClientServiceConnector.cs

```diff
@@ -29,143 +24,42 @@ public class ClientServiceConnector
-  /// <param name="sslValidation">Check if the ssl must have a strong validation</param>
-  /// <param name="loggerFactory">Optional logger factory</param>
-  /// <returns></returns>
-  private static ChannelBase ControlPlaneConnection(string                     endPoint,
-                                                    Tuple<string, string>      clientPem     = null,
-                                                    bool                       sslValidation = true,
-                                                    [CanBeNull] ILoggerFactory loggerFactory = null)
-  {
-    var uri = new Uri(endPoint);
-
-    var credentials = uri.Scheme == Uri.UriSchemeHttps
-                        ? new SslCredentials()
-                        : ChannelCredentials.Insecure;
-    var httpClientHandler = new HttpClientHandler();
-    if (!sslValidation)
-    {
-      httpClientHandler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
-      AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport",
-                           true);
-    }
-
-#if NET5_0_OR_GREATER
-    if (clientPem != null)
-    {
-      var cert = X509Certificate2.CreateFromPem(clientPem.Item1,
-                                                clientPem.Item2);
-
-      // Resolve issue with Windows on pem bug with windows
-      // https://github.com/dotnet/runtime/issues/23749#issuecomment-388231655
-
-      if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-      {
-        var originalCert = cert;
-        cert = new X509Certificate2(cert.Export(X509ContentType.Pkcs12));
-        originalCert.Dispose();
-      }
-
-      httpClientHandler.ClientCertificates.Add(cert);
-    }
-
-    var channelOptions = new GrpcChannelOptions
-                         {
-                           Credentials = uri.Scheme == Uri.UriSchemeHttps
-                                           ? new SslCredentials()
-                                           : ChannelCredentials.Insecure,
-                           HttpHandler = httpClientHandler,
-                           LoggerFactory = loggerFactory,
-                         };
-
-    var channel = GrpcChannel.ForAddress(endPoint,
-                                         channelOptions);
-
-#else
+#if !NET5_0_OR_GREATER
     Environment.SetEnvironmentVariable("GRPC_DNS_RESOLVER",
                                        "native");
-    if (clientPem != null)
-    {
-      credentials = new SslCredentials(clientPem.Item1,
-                                       new KeyCertificatePair(clientPem.Item1,
-                                                              clientPem.Item2));
-    }
-
-    var channel = new Channel($"{uri.Host}:{uri.Port}",
-                              credentials);
 #endif
-    return channel;
+    return GrpcChannelFactory.CreateChannel(new GrpcClient
```

Comment:
Could you run the clientwith Dotnet Frameworks 4.6. I think this one compile but fail at runtime on windows

Reply:
Indeed, even after the changes in Api, .net framework doesn't support http2, so back to the drawing board in API